File: 401-600/e509.java
class e509 {
    public static int fib(int N) {
        if (N == 0){
            return 0;
        } else if (N == 1 || N == 2){
            return 1;
        }
        return fib(N-1) + fib(N-2);
    }

    public static void main(String[] args){
        System.out.println(fib(2)); //1
        System.out.println(fib(3)); //2
        System.out.println(fib(4)); //3
    }
}

File: 401-600/e509.py
class Solution(object):
    def fib(self, N):
        """
        :type N: int
        :rtype: int
        """
        if N == 0:
            return 0
        
        if N in [1, 2]:
            return 1

        a = 0
        b = 1
        for i in range(N):
            a, b = b, a+b
        
        return a
            
